app/Support/PluginManager.php

```
@@ -1589,22 +1595,51 @@ public function log(?int $pluginId, string $level, string $message, array $conte
-        $this->cleanupOrphanPlugins();
-
-        // Get all active plugins
-        $activePlugins = $this->getActivePlugins();
+        // Bundled-plugin registration and orphan cleanup are maintenance
+        // operations: they scan the filesystem and issue several queries, yet
+        // their outcome only changes after an update or an admin plugin action.
+        // Run them at most once per interval (per app version) instead of on
+        // every request; plugin lifecycle methods clear the marker so admin
+        // actions re-trigger the sync immediately.
+        $maintenanceKey = 'plugins_maintenance_' . self::appVersionStamp();
+        if (QueryCache::get($maintenanceKey) === null) {
```

PHPMD (2.15.0)

[error] Avoid using static access to class 'App\Support\QueryCache' in method 'loadActivePlugins'. (undefined)

(StaticAccess)